frontend/containers/layouts/discover-search/component.tsx

```diff
@@ -41,185 +38,143 @@ export const DiscoverSearch: FC<DiscoverSearchProps> = ({
-    Object.entries(filters).forEach(([key, value]) => {
-      // Just send the used filters
-      if (value) {
-        filterParams[`filter[${key}]`] = `${value}`;
-      }
-    });
-    return filterParams;
-  };
-
-  const handleSubmitFilters = (filters: FilterForm) => {
-    setFiltersInputValue(filters);
-    const filterParams = getFilterParams(filters);
-    handleSearch(searchInputValue, filterParams);
+    const { value } = e.currentTarget;
+    if (value?.length >= 1) {
+      setShowSuggestions(true);
+    } else {
+      setShowSuggestions(false);
+    }
+    setSearchInputValue(value);
     setOpenFilters(false);
-    setShowSuggestions(false);
   };
 
-  const handleFilterSuggestion = (filter: Enum) => {
-    const filterKey = `filter[${filter.type}]`;
-    handleSearch('', { ...filters, [filterKey]: filter.id });
-    setFiltersInputValue({ ...filtersInputValue, [filter.type]: filter.id });
-    setShowSuggestions(false);
+  const clearInput = () => {
     setSearchInputValue('');
+    setShowSuggestions(false);
+    doSearch('', filters);
   };
 
-  const handleSearchSuggestion = () => {
-    handleSearch(searchInputValue, filters);
+  const handleClickInput = () => {
+    setOpenFilters(true);
     setShowSuggestions(false);
   };
 
+  const showActiveFilters = useMemo(() => !!values(filters).length, [filters]);
+
   return (
     <div className={className}>
       <div
-        className={cx('z-10 relative w-full sm:h-16 text-black bg-white border drop-shadow-xl', {
-          'rounded-full': !openFilters && !showSuggestion,
-          'rounded-t-4xl': openFilters || showSuggestion,
-        })}
+        className={cx(
+          'relative z-10 w-full sm:overflow-visible text-black bg-white border h-14 sm:h-16 drop-shadow-xl rounded-3xl',
+          {
+            'rounded-b-none mb-0': showSuggestion || openFilters || showActiveFilters,
+          }
+        )}
       >
         <div
-          className={cx('flex items-center justify-between px-6 py-3 sm:gap-4', {
+          className={cx('flex h-full items-center justify-between px-4 sm:px-6 py-2 sm:gap-4', {
             'py-4': openFilters || showSuggestion,
           })}
         >
           <form
             role="search"
-            className="flex flex-col items-center justify-end w-full h-full gap-1 sm:justify-between sm:gap-3 sm:flex-row"
+            className="flex items-center justify-end w-full h-full gap-2 sm:justify-between sm:gap-3"
             onSubmit={handleSubmitSearch}
           >
             <div className="flex items-center flex-grow gap-2">
-              <Icon
-                aria-hidden={true}
-                icon={SearchIcon}
-                className="w-6 h-6 mr-2 sm:w-8 sm:h-8 text-green-dark"
-              />
+              <Button
+                theme="naked"
+                size="smallest"
+                className="hidden px-3 py-3 sm:block"
+                onClick={clearInput}
```

Comment:
It's not really clear to me why the magnifying glass icon would empty the search input. Is that something Andreia asked for?

Reply:
No, it should open the filters. I fixed it